exporter/signozkafkaexporter/utils.go

```diff
@@ -15,3 +17,24 @@ func getKafkaTopicFromClientMetadata(md client.Metadata) (string, error) {
 	}
 	return signozTenantId[0], nil
 }
+
+// getKafkaTopicFromClientMetadata returns the kafka topic from client metadata
+func getRequestIdFromClientMetadata(md client.Metadata) (string, error) {
+	// return default topic if no tenant id is found in client metadata
+	signozTenantId := md.Get("signoz_request_id")
+	if len(signozTenantId) == 0 {
+		return "", fmt.Errorf("signoz_request_id not found in client metadata")
+	}
+	return signozTenantId[0], nil
+}
+
+func setRequestIdAsKey(ctx context.Context, messages []*sarama.ProducerMessage) ([]*sarama.ProducerMessage, error) {
+	requestId, err := getRequestIdFromClientMetadata(client.FromContext(ctx).Metadata)
+	if err != nil {
+		return nil, err
+	}
+	for _, message := range messages {
+		message.Key = sarama.StringEncoder(requestId)
+	}
```

Comment:
Why is it assigning the same identifier to each item here? Would that result in losing the legitimate data? Shouldn't there be sequencing here?

Reply:
Based on the assumption stated here https://github.com/SigNoz/signoz-otel-collector/pull/273/files#diff-06a5e74f9ab7315cfd07374023d50c6ccf75d45afbf188f0047c76dfaf4ce968R52 we would lose data. We should sequence them or if the assumption is not correct, then update the comment and panic when the size is not 1.